app/src/main/java/tech/relaycorp/courier/domain/client/CargoDelivery.kt

```diff
@@ -24,16 +27,18 @@ class CargoDelivery
     private val deleteMessage: DeleteMessage
 ) {
 
-    suspend fun deliver() {
+    suspend fun deliver(dohClient: DoHClient) {
         getCargoesToDeliver()
             .groupByRecipient()
             .forEach { (recipientAddress, cargoes) ->
                 try {
-                    deliverToRecipient(recipientAddress, cargoes)
+                    deliverToRecipient(recipientAddress as PublicMessageAddress, cargoes, dohClient)
                 } catch (e: IncompleteDeliveryException) {
-                    logger.log(Level.WARNING, "Cargo delivery error", e)
+                    logger.log(Level.WARNING, "Server failed to acknowledge one or more cargoes")
```

Comment:
Unrelated to this PR, but I got confused by the stack trace and thought it shouldn't be there.